src/mcp/client/streamable_http.py

```diff
@@ -102,6 +115,7 @@ def __init__(
             CONTENT_TYPE: JSON,
             **self.headers,
```

Comment:
BTW I would expect any auth headers passed in `self.headers` to take precedence over the `auth_provider` headers - in many APIs explicitly setting a value takes precedence over implicitly inferring it (e.g. I've seen that pattern often in MLflow), but would defer to the maintainers' opinion on that

Reply:
Good call, I added the behaviour to not override passed in headers, and add a test case as well